kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesCrudDispatcher.java

```diff
@@ -42,6 +71,50 @@ public KubernetesCrudDispatcher(AttributeExtractor attributeExtractor, ResponseC
     super(new Context(Serialization.jsonMapper()), attributeExtractor, responseComposer);
   }
 
+  @Override
+  public MockResponse dispatch(RecordedRequest request) {
+    String path = request.getPath();
+    String method = request.getMethod();
+    switch (method.toUpperCase()) {
+      case POST:
+      case PUT:
+        return handleCreate(path, request.getBody().readUtf8());
+      case PATCH:
+        return handlePatch(path, request.getBody().readUtf8());
+      case GET:
+        return detectWatchMode(path)? handleWatch(request): handleGet(path);
+      case DELETE:
+        return handleDelete(path);
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Adds the specified object to the in-memory db.
+   *
+   * @param path
+   * @param s
+   * @return
+   */
+  public MockResponse handleCreate(String path, String s) {
+    Serialization.unmarshal(s, KubernetesResource.class);
+    MockResponse response = new MockResponse();
+    AttributeSet features = AttributeSet.merge(attributeExtractor.fromPath(path), attributeExtractor.fromResource(s));
+    map.put(features, s);
+
+    for (Map.Entry<AttributeSet, Set<WatchEventsListener>> entry: watchEventListenerMapping.entrySet()) {
+      if (features.matches(entry.getKey())) {
```

Comment:
The key object should be improved or replaced so that you don't have to iterate through the Map's entries and check if the key `matches`.

If your key had a proper equals/hashCode, you could simply get the value from the map `watchEventListenerMapping.get(features)`

Just create an extra object for the Key that encapsulates the AttributeSet and whatever other thing you're using to identify the object and implement the equals/hashCode methods.